### ISSUE TITLE
Compile & Release RC/M Version Against Scala 3

### ISSUE THREAD

**Issue by leedsalex** [tminglei/slick-pg#657]
Can you please compile the latest `0.22.0-M4` version against Scala 3 and release to maven? Slick has [done the same](https://mvnrepository.com/artifact/com.typesafe.slick/slick_3/3.5.0-M4).

**Comment by mdedetrich** [comment]
This is going to be non trivial to do because slick-pg heavily relies on reflection which is removed in Scala 3. I am currently doing a PR to move from scala reflection to [izumi-reflect](https://github.com/zio/izumi-reflect) which supports all major Scala versions

**Comment by hughsimpson** [comment]
@dejvid Yeah I've tried that with [this branch](https://github.com/tminglei/slick-pg/pull/677) and it works fine. Might be good if you're trying to fix scala 3 compatibilities before this is published
